### ISSUE TITLE
Error: MiniDumpWriteDump failed with code 8007012b

### ISSUE THREAD

**Issue by cclements** [hoangprod/AndrewSpecial#2]
Hey there, fun little PoC!  After compiling and running against Windows 10 1809 x64, I get the following error:
```
AndrewSpecial.exe
RPM: 75905440 ---- ntRVM: 00B016BA
ntReadVirtualMemory Syscall is 3f
Got lsass.exe handle: 12c
Error: MiniDumpWriteDump failed with code 8007012b
```
The exe works as expected against Win7 x32 SP1

Is there anything I need to modify to support my version of Win 10?

**Comment by hoangprod** [comment]
Oh I am so dumb, I forgot to utilize the 32bit syscall id (https://j00ru.vexillium.org/syscalls/nt/32/) and only uses 64bit. Replace 3f with the correct version of your windows 10 syscall NtReadVirtualMemory and you should be good. I'll be honestly, it shouldn't work at all on 32bit versions....

**Comment by xforjustice** [comment]
Hi,
I am getting similar error running it on Windows 10 1803 x64. The syscall id seems right '3f' but still after compiling and running it on the same host I get the following error:
C:\Users\hpj675\Desktop\AndrewSpecial-master>AndrewSpecial.exe
RPM: 75B35B70 ---- ntRVM: 013716CA
ntReadVirtualMemory Syscall is 3f
Got lsass.exe handle: 114
Error: MiniDumpWriteDump failed with code 8007012b

Anything I am missing ?

**Comment by hoangprod** [comment]
Is this on a box with or without protection? Looking up the error code, it seem to be relating to RPM not being able to complete. Can you make sure that the HANDLE that AndrewSpecial.exe opened have all the permissions? You can check this using tool such as ProcessHacker. I would also recommend running the same binary on a unprotected box to see if the current EDR solution is not interfering with the execution, if one is currently present ofc.

**Comment by xforjustice** [comment]
There are no protection on the box. And other tools like sqldumper and procdump works. I ran the PoC as system.

**Comment by wshenk** [comment]
I am having the same problem.  

OS Name:                   Microsoft Windows 10 Enterprise 2015 LTSB
OS Version:                10.0.10240 N/A Build 10240

wmic os get OSArchitecture
OSArchitecture
64-bit

Error: MiniDumpWriteDump failed with code 8007012b

**Comment by tomwillow** [comment]
I got the same error code, and I found a reason. Maybe your situation is same as me.

After call MiniDumpWriteDump, this struct may be created:

  MINIDUMP_EXCEPTION_INFORMATION mdei = {0};
  mdei.ThreadId = GetCurrentThreadId();
  mdei.ExceptionPointers = nullptr; // <- if here is NULL, the ClientPointers must be FALSE
  mdei.ClientPointers = FALSE;

If ExceptionPointers is NULL and ClientPointers is TRUE, this error code 0x8007012b will be returned at MiniDumpWriteDump.
